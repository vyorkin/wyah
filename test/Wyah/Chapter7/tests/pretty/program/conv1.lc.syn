Program [Decl "church" (EFix (ELam (Var "church") (ELam (Var "n") (EIf (EOp Eq (EVar (Var "n")) (ELit (LInt 0))) (EVar (Var "zero")) (ELam (Var "f") (ELam (Var "x") (EApp (EVar (Var "f")) (EApp (EApp (EApp (EVar (Var "church")) (EOp Sub (EVar (Var "n")) (ELit (LInt 1)))) (EVar (Var "f"))) (EVar (Var "x")))))))))),Decl "unchurch" (ELam (Var "n") (EApp (EApp (EVar (Var "n")) (ELam (Var "i") (EOp Add (EVar (Var "i")) (ELit (LInt 1))))) (ELit (LInt 0)))),Decl "unbool" (ELam (Var "n") (EApp (EApp (EVar (Var "n")) (EVar (Var "True"))) (EVar (Var "False"))))]
