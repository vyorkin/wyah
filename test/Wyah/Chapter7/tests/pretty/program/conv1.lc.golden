let church = fix (\church -> \n -> if n == 0 then zero else \f -> \x -> f (((church (n - 1)) f) x));
let unchurch = \n -> (n (\i -> i + 1)) 0;
let unbool = \n -> (n True) False;